File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    # 문자열은 알파벳 소문자
    # 문자열에서 같은 알파벳 두개가 연속해질때마다 그 둘을 제거
    
    stack = []
    for i in s:
        if len(stack)>0 and stack[-1] == i:
            stack.pop()
        else:
            stack.append(i)
    
    if len(stack) == 0:
        return 1
    if len(stack)%2 != 0:
        return 0
    while len(stack)>0:
        if stack.pop() != stack[-1]:
            return 0
        stack.pop()

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    stack<char> st;
    st.push(s[0]);
    int idx = 1;
    while (idx < s.size()) {
        if (!st.empty() && st.top() == s[idx]) {
            st.pop();
        } else {
            st.push(s[idx]);
        }
        idx++;
    }
    return st.empty();
}